tests/ui_tools/test_boxes.py

```diff
@@ -340,6 +340,68 @@ def test_generic_autocomplete_mentions_subscribers(
         typeahead_string = write_box.generic_autocomplete(text, state)
         assert typeahead_string == required_typeahead
 
+    @pytest.mark.parametrize(
+        "text",
+        [
+            prefix + "Human"[: index + 1]
+            for index in range(len("Human"))
+            for prefix in ["@", "@**"]
+        ]
+        + ["@**"],
+    )
+    def test_generic_autocomplete_user_mentions(self, write_box, text, mocker, state=1):
+        _process_typeaheads = mocker.patch(BOXES + ".WriteBox._process_typeaheads")
+
+        matching_users = [
+            "Human Myself",
+            "Human 1",
+            "Human 2",
+            "Human Duplicate",
+            "Human Duplicate",
+        ]
+        distinct_matching_users = [
+            "@**Human Myself**",
+            "@**Human 1**",
+            "@**Human 2**",
+            "@**Human Duplicate|13**",
+            "@**Human Duplicate|14**",
+        ]
+
+        write_box.generic_autocomplete(text, state)
+
+        _process_typeaheads.assert_called_once_with(
+            distinct_matching_users, state, matching_users
+        )
+
+    @pytest.mark.parametrize(
+        "text", ["@_" + "Human"[:index] for index in range(len("Human") + 1)]
+    )
+    def test_generic_autocomplete_silent_user_mentions(
+        self, write_box, text, mocker, state=1
+    ):
+        _process_typeaheads = mocker.patch(BOXES + ".WriteBox._process_typeaheads")
+
+        matching_users = [
+            "Human Myself",
+            "Human 1",
+            "Human 2",
+            "Human Duplicate",
+            "Human Duplicate",
+        ]
+        distinct_matching_users = [
+            "@_**Human Myself**",
+            "@_**Human 1**",
+            "@_**Human 2**",
+            "@_**Human Duplicate|13**",
+            "@_**Human Duplicate|14**",
+        ]
```

Comment:
This test seems the same as the previous but with a different prefix; can we not just add that to the parametrize?